src/app/CommandHandler.h

```diff
@@ -424,293 +171,77 @@ class CommandHandler
-     * @param payload Payload of the incoming InvokeRequestMessage from the client.
-     * @param isTimedInvoke Indicates whether the interaction is timed.
-     * @param faultType The specific type of fault to inject into the response.
+     * @return The inner exchange context, might be nullptr if no
+     *         exchange context has been assigned or the context
+     *         has been released.
      */
-    // TODO(#30453): After refactoring CommandHandler for better unit testability, create a
-    // unit test specifically for the fault injection behavior.
-    void TestOnlyInvokeCommandRequestWithFaultsInjected(CommandHandlerExchangeInterface & commandResponder,
-                                                        System::PacketBufferHandle && payload, bool isTimedInvoke,
-                                                        NlFaultInjectionType faultType);
-#endif // CHIP_WITH_NLFAULTINJECTION
+    virtual Messaging::ExchangeContext * GetExchangeContext() const = 0;
 
-private:
-    friend class TestCommandInteraction;
-    friend class CommandHandler::Handle;
-
-    enum class State : uint8_t
-    {
-        Idle,                ///< Default state that the object starts out in, where no work has commenced
-        NewResponseMessage,  ///< mInvokeResponseBuilder is ready, with no responses added.
-        Preparing,           ///< We are prepaing the command or status header.
-        AddingCommand,       ///< In the process of adding a command.
-        AddedCommand,        ///< A command has been completely encoded and is awaiting transmission.
-        DispatchResponses,   ///< The command response(s) are being dispatched.
-        AwaitingDestruction, ///< The object has completed its work and is awaiting destruction by the application.
-    };
+    // actual impls
 
     /**
-     * @brief Best effort to add InvokeResponse to InvokeResponseMessage.
-     *
-     * Tries to add response using lambda. Upon failure to add response, attempts
-     * to rollback the InvokeResponseMessage to a known good state. If failure is due
-     * to insufficient space in the current InvokeResponseMessage:
-     *  - Finalizes the current InvokeResponseMessage.
-     *  - Allocates a new InvokeResponseMessage.
-     *  - Reattempts to add the InvokeResponse to the new InvokeResponseMessage.
+     * API for adding a data response.  The template parameter T is generally
+     * expected to be a ClusterName::Commands::CommandName::Type struct, but any
+     * object that can be encoded using the DataModel::Encode machinery and
+     * exposes the right command id will work.
      *
-     * @param [in] addResponseFunction A lambda function responsible for adding the
-     *             response to the current InvokeResponseMessage.
+     * @param [in] aRequestCommandPath the concrete path of the command we are
+     *             responding to.
+     * @param [in] aData the data for the response.
```

Comment:
My thinking was that I have no better documentation than `CHIP_NO_ERROR on success, error code otherwise`  (also I believe it was a copy & paste) which seems not that useful. Will try to see if I can make the documentation useful